### ISSUE TITLE
tslint: Assets parent identified as different Construct type

### ISSUE THREAD

**Issue by costleya** [awslabs/aws-cdk#1137]
```typescript
const buildInstructions = new FileAsset(this, 'DotnetDevexPipelineBuildInstructions', {
    path: './content/build.sh'
});
```
yields

```
Type 'DotnetDevexPipelineStack' is not assignable to type 'Construct'.
    Types of property 'parent' are incompatible.
      Type 'import("/Users/costleya/Work/dotnet-devex-pipeline/node_modules/@aws-cdk/cdk/lib/core/construct").Construct | undefined' is not assignable to type 'import("/Users/costleya/Work/dotnet-devex-pipeline/node_modules/@aws-cdk/assets/node_modules/@aws-cdk/cdk/lib/core/construct").Construct | undefined'.
        Type 'import("/Users/costleya/Work/dotnet-devex-pipeline/node_modules/@aws-cdk/cdk/lib/core/construct").Construct' is not assignable to type 'import("/Users/costleya/Work/dotnet-devex-pipeline/node_modules/@aws-cdk/assets/node_modules/@aws-cdk/cdk/lib/core/construct").Construct'.
          Types have separate declarations of a private property '_children'.
```

**Comment by eladb** [comment]
This is supposed to be fixed in >0.15.2 as we introduced peer dependencies (#1091, #979). The problem is not specific to assets, it's because you have two versions of @aws-cdk/cdk in your closure and TypeScript won't allow you to interchangeably use them.

Closing for now, let us know if this persists with the latest version

**Comment by codypenta** [comment]
To elaborate on this error in case someone reached this page later like I did. While you're developing in typescript and adding new services you may run into an issue where all of your aws-cdk/aws-{someservice} is running on 1.6.0. But the new package you installed will be version 1.6.1 and will give you an error every time you're using "this." Easy fix with npm upgrade command
